### PR TITLE
Swap some Gasoline pumps for Diesel at Rural & Gas stations

### PR DESCRIPTION
**Pull request by meladath**
SUMMARY: Features "Swap half of the Gasoline pumps at Rural & Gas stations for Diesel pumps"
#### Purpose of change

Diesel fuel (in pumps) seems almost impossible to find out in the world for some strange reason. It is abundant in real life and considering how many truckers stop at gas stations, the absence of diesel pumps at both automated gas stations and rural gas stations is a complete mystery. I mean, doesnt pretty much every station in real life have both gasoline and diesel pumps?

#### Describe the solution

I have replaced half of the Gasoline pumps at rural gas stations (1 of 2) and gas stations (3 of 6)  with Diesel pumps. I also attempted to retain compatibility with the automated fuel console, however it seems this was broken to begin with for Gasoline pumps anyway.

#### Testing

Spawned both types of tile and attempted to use the pumps, they function as intended. Attempted to use the automated gas console to "buy" both gas and diesel, tanks are correctly seperate with seperate values however automated console doesnt seem to refill the pump (this is the same functionality as before this PR?)

### CODE REVIEW

File: src/iexamine.cpp [hunk 1]
```diff
@@ -4052,23 +4077,24 @@ void iexamine::pay_gas( player &p, const tripoint &examp )
         popup( _( "You're illiterate, and can't read the screen." ) );
     }
 
-    int pumpCount = getNearPumpCount( examp );
+    std::string fuelType;
+    int pumpCount = getNearPumpCount( examp, fuelType );
     if( pumpCount == 0 ) {
-        popup( str_to_illiterate_str( _( "Failure!  No gas pumps found!" ) ) );
+        popup( str_to_illiterate_str( _( "Failure!  No " + fuelType + " pumps found!" ) ) );
```
Comment (ZhilkinSerg) [inline src/iexamine.cpp:4083]: That probably won't be translated. You probably need to use something like this:

```suggestion
        popup( str_to_illiterate_str( string_format( _( "Failure!  No %s pumps found!"), fuelType ) ) );
```

Comment (meladath) [reply src/iexamine.cpp:4083]: Thank you, will fix this now.

**Comment by meladath** [comment]
Can someone check the addition of str_to_illiterate_str on the purchase message please? It feels like it was simply missing all along, but maybe there was a reason that specific string wasnt passed through str_to_illiterate_str?

File: src/iexamine.cpp [hunk 2]
```diff
@@ -4141,10 +4173,10 @@ void iexamine::pay_gas( player &p, const tripoint &examp )
             return;
         }
 
-        int maximum_liters = std::min( money / pricePerUnit, tankGasUnits / 1000 );
+        int maximum_liters = std::min( money / pricePerUnit, tankUnits / 1000 );
 
-        std::string popupmsg = string_format(
-                                   _( "How many liters of gasoline to buy?  Max: %d L.  (0 to cancel)" ), maximum_liters );
+        std::string popupmsg = str_to_illiterate_str( string_format(
```
Comment (meladath) [inline src/iexamine.cpp:4178]: Can someone check the addition of str_to_illiterate_str on the purchase message please? It feels like it was simply missing all along, but maybe there was a reason that specific string wasnt passed through str_to_illiterate_str?

File: src/iexamine.cpp [hunk 3]
```diff
@@ -3840,27 +3840,41 @@ void iexamine::sign( player &p, const tripoint &examp )
     }
 }
 
-static int getNearPumpCount( const tripoint &p )
+static int getNearPumpCount( const tripoint &p, std::string &fuel_type )
```
Comment (ZhilkinSerg) [inline src/iexamine.cpp:3843]:
```suggestion
static int getNearPumpCount( const tripoint &p, const std::string &fuel_type )
```

File: src/iexamine.cpp (hunk 2 above)
Comment (ZhilkinSerg) [reply src/iexamine.cpp:4178]: Hard to say - that was 6 years ago (https://github.com/CleverRaven/Cataclysm-DDA/pull/8137).

**Comment by Mister-Cultist** [comment]
would have thought almost every station should have a diesal pump. Good job making diesal cars more viable.

**Pull request merged**